packages/langium/src/grammar/langium-grammar-semantic-token-provider.ts

```diff
@@ -0,0 +1,32 @@
+/******************************************************************************
+ * Copyright 2022 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { SemanticTokenTypes } from 'vscode-languageserver';
+import { AbstractSemanticTokenProvider } from '../lsp/semantic-token-provider';
+import { AstNode } from '../syntax-tree';
+import { isAction, isAssignment, isParameter, isParameterReference, isParserRule, isTerminalRule } from './generated/ast';
+import { isDataTypeRule } from './grammar-util';
+
+export class LangiumGrammarSemanticTokenProvider extends AbstractSemanticTokenProvider {
+
+    protected highlightElement(node: AstNode): boolean {
+        if (isAssignment(node)) {
+            this.highlightFeature(node, 'feature', SemanticTokenTypes.property);
+        } else if (isAction(node)) {
+            if (node.feature) {
+                this.highlightFeature(node, 'feature', SemanticTokenTypes.property);
+            }
+        } else if ((isParserRule(node) && isDataTypeRule(node) || isTerminalRule(node)) && node.type) {
+            this.highlightFeature(node, 'type', SemanticTokenTypes.type);
```

Comment:
This seems not to work for me. It's supposed to highlight primitive return types, right?

Reply:
It works on my machine 🤔

![image](https://user-images.githubusercontent.com/4377073/148952326-955623a3-806e-4b7a-bf0e-6a4d56e66c8f.png)

Reply:
Now it works here as well 🤷‍♂️